templates/rollup-template/src/rollup.rs

```diff
@@ -0,0 +1,100 @@
+use serde::de::DeserializeOwned;
+use sov_db::ledger_db::LedgerDB;
+use sov_modules_api::default_context::{DefaultContext, ZkDefaultContext};
+use sov_modules_stf_template::AppTemplate;
+use sov_rollup_interface::services::da::DaService;
+use sov_rollup_interface::zk::ZkvmHost;
+use sov_stf_runner::{Prover, RollupConfig, RunnerConfig, StateTransitionRunner};
+use tokio::sync::oneshot;
+
+use crate::builder::StfWithBuilder;
+use crate::rpc::{register_ledger, register_sequencer};
+use crate::runtime::{get_rpc_methods, GenesisConfig, Runtime};
+
+type ZkStf<Da, Vm> = AppTemplate<ZkDefaultContext, Da, Vm, Runtime<ZkDefaultContext, Da>>;
+
+/// Dependencies needed to run the rollup.
+// This is duplicated exactly from demo-rollup. Should go to stf-runner crate?
+pub struct Rollup<Vm: ZkvmHost, Da: DaService + Clone> {
```

Comment:
It looks like the `new`, `run_and_report_rpc_port`, and the `jsonrpsee::RpcModule` registration logic is common for any rollup.

Would it make sense to introduce:
```rust
struct RollupTemplate <GenesiConfig, R: Runtime>{
     pub runtime: R,
     pub da_service: Da,
     pub ledger_db: LedgerDB,
     pub runner_config: RunnerConfig,
     pub genesis_config: GenesisConfig,
     #[allow(clippy::type_complexity)]
     pub prover: Option<Prover<ZkStf<Da::Spec, Vm::Guest>, Da, Vm>>,
     ///======  This is new
     methods: Vet<jsonrpsee::RpcModule<()>
} 
```
And implement `run_and_report_rpc_port ` generically (we could merge all the `methods` in a loop).
This structure would have a similar rolle to `AppTemplate`. 
`AppTemplate` gives implementation of STF for free
`RollupTemplate` gives implementation of a Rollup for free 
They could leave in the same module. 
The `demo-rollup` & `this package (we would need to change the name)` would be an instantiation of `RollupTemplate ` with different dependencies.

Reply:
Yeah, this definitely makes sense. Planning to do this in the next PR